src/ngraph/runtime/tensor.cpp

```diff
@@ -95,3 +97,41 @@ void runtime::Tensor::copy_from(const ngraph::runtime::Tensor& source)
     source.read(buffer.get_ptr(), size);
     write(buffer.get_ptr(), size);
 }
+
+future<void> runtime::Tensor::begin_write(const void* p, size_t size_in_bytes, size_t buffer_number)
+{
+    if (m_backend)
+    {
+        // auto f = m_promise.get_future();
+        return m_backend->post_async_write_event(
+            shared_from_this(), p, size_in_bytes, buffer_number);
+    }
+    else
+    {
+        throw runtime_error("Async operations not supported for this backend");
```

Comment:
What's the caller supposed to do if async isn't supported?

Reply:
Contact the backend developer and get them to support it. In the future we can make this API required but for now it is optional.